packages/components/src/internal/components/forms/input/AssayTaskInput.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { FC, memo, useCallback, useEffect, useReducer } from 'react';
+import { Filter } from '@labkey/api';
+
+import { Alert, LoadingSpinner, SelectInput, selectRows } from '../../../..';
+
+interface InputOption {
+    label: string;
+    value: number;
+}
+
+async function loadInputOptions(assayId: number): Promise<InputOption[]> {
+    const { key, models } = await selectRows({
+        schemaName: 'samplemanagement',
+        queryName: 'Tasks',
+        columns: 'RowId,Name,AssayTypes,Run/Name',
+        filterArray: [
+            Filter.create('AssayTypes', undefined, Filter.Types.NONBLANK),
+            Filter.create('Status/Value', 'In Progress'),
+        ],
+        maxRows: -1,
+    });
+    const rows = Object.values(models[key]);
+    const taskOptions = [];
+
+    rows.forEach(row => {
+        const taskId = row['RowId'].value;
+        const jobName = row['Run/Name'].value;
+        const taskName = row['Name'].value;
+        const assays = row['AssayTypes'].value.split(',').map(Number);
+        const hasAssay = assays.find(id => assayId === id);
+
+        if (hasAssay) {
+            taskOptions.push({ label: `${jobName} - ${taskName}`, value: taskId });
+        }
+    });
+
+    return taskOptions;
+}
+
+interface WorkflowTaskInputOptionsState {
+    error: string;
+    loading: boolean;
+    taskOptions: InputOption[];
+    value: number;
+}
+
+interface WorkflowTaskInputHook {
+    setValue: (value: number) => void;
+    state: WorkflowTaskInputOptionsState;
+}
+
+const DEFAULT_STATE = {
+    taskOptions: undefined,
+    value: undefined,
+    loading: true,
+    error: undefined,
+};
+
+const inputReducer = (
+    state: WorkflowTaskInputOptionsState,
+    change: Partial<WorkflowTaskInputOptionsState>
+): WorkflowTaskInputOptionsState => ({ ...state, ...change });
+
+function useWorkflowTaskInputState(assayId: number, value?: number): WorkflowTaskInputHook {
+    const [state, setState] = useReducer(inputReducer, { ...DEFAULT_STATE, value });
+    const load = async (): Promise<void> => {
+        if (assayId === undefined) {
+            // If the components rendering the QueryFormInputs or EditableDetailPanel don't properly inject the assayId
+            // into the form data (via ASSAY_INDEX key defined above) then this will happen.
+            setState({ loading: false, error: 'Assay ID not set, cannot load Workflow Tasks' });
```

Comment:
```suggestion
            setState({ loading: false, error: 'Assay ID not set; cannot load workflow tasks' });
```
I'm not sure we need to use "workflow tasks" as a proper noun.